### DISCUSSION TITLE
ReferenceError: moment is not defined in Vite with Laravel

### DISCUSSION THREAD

**Discussion by csteri** [moment/moment#6422] [Q&A] [answered]
I'm trying to import the moment library via npm install in my laravel project that builds with vite.js
In the console I get this error: **ReferenceError: moment is not defined**

vite.config.js
``` js
import { defineConfig } from 'vite';
import laravel from 'laravel-vite-plugin';

import moment from 'moment';

export default defineConfig({
    plugins: [
        laravel([
            'resources/css/app.css',
            'resources/js/app.js',
        ]),
    ],
});
```

package.json
```js
{
    "private": true,
    "scripts": {
        "dev": "vite",
        "build": "vite build"
    },
    "devDependencies": {
        "axios": "^0.25",
        "laravel-vite-plugin": "^0.2.1",
        "lodash": "^4.17.19",
        "postcss": "^8.1.14",
        "sass": "^1.53.0",
        "vite": "^2.9.11"
    },
    "dependencies": {
        "@fortawesome/fontawesome-free": "^6.1.1",
        "@popperjs/core": "^2.11.5",
        "bootstrap": "^5.1.3",
        "bootstrap-icons": "^1.9.0",
        "moment": "^2.29.4"
    }
}

```

in welcome.blade.php

```html
<head>
...
@vite('resources/js/app.js')
</head>

<body>
   <script>
      console.log(moment());
   </script>
</body>
```

after running `npm run dev`, everything is ok, I have no errors

**Desktop:**
 - OS: Windows 10
 - Browser: Chrome Version 103.0.5060.134
 - Version laravel: 9.19
 - Version moment: 2.29.4

**Comment by csteri** [comment]
I put `"moment": "^2.29.4"` in `package.json -> 'devDependencies': { }` and it worked.

But I get a warning in the console: `Could not load content for https://momentjs.com/downloads/moment.min.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE`

**Comment by mattjohnsonpint** [comment, marked as answer]
Thanks for the detailed setup! Importing Moment in `vite.config.js` exposes it only to Vite's Node configuration, not to browser code. I'd import it in `resources/js/app.js`. If legacy inline scripts truly require a global, I'd explicitly assign `window.moment = moment` there. I'm closing this as answered; the import was simply on the wrong side of the bundler.
